diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2d61aff1..74802e95 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -265,30 +265,8 @@ set(COOLPROP_INCLUDE_DIRECTORIES ${APP_INCLUDE_DIRS} CACHE LIST "List of include
 #######################################
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/dev/cmake/Modules/")
 
-set(Python_ADDITIONAL_VERSIONS 2.7 2.6 2.5 2.4)
-find_package (PythonInterp 2.7)
-if (NOT PYTHON_EXECUTABLE)
-  MESSAGE(STATUS "Looking for Python")
-  find_package (Python COMPONENTS Interpreter)
-endif()
-if (NOT PYTHON_EXECUTABLE)
-  MESSAGE(STATUS "Looking for Python2")
-  find_package (Python2 COMPONENTS Interpreter)
-  if(Python2_Interpreter_FOUND)
-    set(PYTHON_EXECUTABLE ${Python2_EXECUTABLE})
-  endif()
-endif()
-if (NOT PYTHON_EXECUTABLE)
-  MESSAGE(STATUS "Looking for Python3")
-  find_package (Python3 COMPONENTS Interpreter)
-  if(Python3_Interpreter_FOUND)
-    set(PYTHON_EXECUTABLE ${Python3_EXECUTABLE})
-  endif()
-endif()
-if (NOT PYTHON_EXECUTABLE)
-  MESSAGE(WARNING "Could not find Python, be prepared for errors.")
-endif()
-
+find_package (Python COMPONENTS Interpreter REQUIRED)	
+set(PYTHON_EXECUTABLE "${Python_EXECUTABLE}")
 if(CMAKE_DL_LIBS)
     find_package (${CMAKE_DL_LIBS} REQUIRED)
 endif()
