From ea5499419c87364b691b1303154575be4f2cb8d3 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Fri, 30 May 2025 09:17:04 +0200
Subject: [PATCH] Update setup.py

---
 wrappers/Python/setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wrappers/Python/setup.py b/wrappers/Python/setup.py
index 60665bd23..f623fa902 100644
--- a/wrappers/Python/setup.py
+++ b/wrappers/Python/setup.py
@@ -427,7 +427,7 @@ def find_cpp_sources(root=os.path.join('..', '..', 'src'), extensions=['.cpp'],
         os.path.join(CProot),
         os.path.join(CProot, 'include'),
         os.path.join(CProot, 'src'),
-        os.path.join(CProot, 'externals', 'Eigen'),
+        os.path.join(os.environ['PREFIX'], 'include', 'eigen3'),
         # os.path.join(CProot, 'externals', 'fmtlib'),  # should be deprecated
         os.path.join(CProot, 'externals', 'fmtlib', 'include'),
         os.path.join(CProot, 'boost_CoolProp'),
