From 08d076ece6bfe70670aaff5c05837260ddf7a07e Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Fri, 30 May 2025 09:22:25 +0200
Subject: [PATCH] Update setup.py

---
 wrappers/Python/setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wrappers/Python/setup.py b/wrappers/Python/setup.py
index 60665bd23..76be2e4fc 100644
--- a/wrappers/Python/setup.py
+++ b/wrappers/Python/setup.py
@@ -22,7 +22,7 @@ def copy2(old, new):
     for jsonfile in glob.glob(os.path.join('CoolProp', 'include', '*_JSON.h')):
         print('removing', jsonfile)
         os.remove(jsonfile)
-    copytree(os.path.join(CProot, 'externals/fmtlib/include/fmt'), os.path.join('CoolProp', 'include', 'fmt'))  # Should be deprecated
+    copytree(os.path.join(os.environ['PREFIX'], 'include', 'fmt'), os.path.join('CoolProp', 'include', 'fmt'))  # Should be deprecated
     #copytree(os.path.join(CProot, 'externals/fmtlib/include/fmt'), os.path.join('CoolProp','include','fmt'))
     copy2(os.path.join(CProot, 'CoolPropBibTeXLibrary.bib'), os.path.join('CoolProp', 'CoolPropBibTeXLibrary.bib'))
     
