From 1495282729a2a81ed6af87a5c95329e97309e8fa Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Fri, 30 May 2025 09:29:36 +0200
Subject: [PATCH] Update setup.py

---
 wrappers/Python/setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wrappers/Python/setup.py b/wrappers/Python/setup.py
index 60665bd23..03fb5dc0c 100644
--- a/wrappers/Python/setup.py
+++ b/wrappers/Python/setup.py
@@ -431,7 +431,7 @@ def find_cpp_sources(root=os.path.join('..', '..', 'src'), extensions=['.cpp'],
         # os.path.join(CProot, 'externals', 'fmtlib'),  # should be deprecated
         os.path.join(CProot, 'externals', 'fmtlib', 'include'),
         os.path.join(CProot, 'boost_CoolProp'),
-        os.path.join(CProot, 'externals', 'msgpack-c', 'include')]
+        os.path.join(os.environ['PREFIX'], 'include')]
 
     # If the file is run directly without any parameters, clean, build and install
     if len(sys.argv) == 1:
