diff --git a/wrappers/Python/setup.py b/wrappers/Python/setup.py
index 5c339c09..61f9d663 100644
--- a/wrappers/Python/setup.py
+++ b/wrappers/Python/setup.py
@@ -208,6 +208,9 @@ if __name__ == '__main__':
                 cmake_config_args += ['-DFORCE_BITNESS_64=ON']
             else:
                 raise ValueError('cmake_bitness must be either 32 or 64; got ' + cmake_bitness)
+        elif cmake_compiler == 'nmake':
+            cmake_build_args = ['--config', '"Release"']
+            cmake_config_args += ['-G', '"NMake Makefiles"']
         elif cmake_compiler == 'default':
             cmake_config_args = []
             if sys.platform.startswith('win'):
@@ -256,7 +259,7 @@ if __name__ == '__main__':
 
         # Now find the static library that we just built
         static_libs = []
-        for search_suffix in ['Release/*.lib', 'Release/*.a', 'Debug/*.lib', 'Debug/*.a', '*.a']:
+        for search_suffix in ['Release/*.lib', 'Release/*.a', 'Debug/*.lib', 'Debug/*.a', '*.a', '*.lib']:
             static_libs += glob.glob(os.path.join(cmake_build_dir, search_suffix))
 
         if len(static_libs) != 1:
